apps/src/code-studio/progressRedux.js

```diff
@@ -84,18 +85,65 @@ const initialState = {
 /**
  * Thunks
  */
+
+// The user has navigated to a new level in the current lesson,
+// so we should update the browser and also set this as the new
+// current level.
 export function navigateToLevelId(levelId) {
   return (dispatch, getState) => {
     const state = getState().progress;
-    const newLevel = state.lessons[0].levels.find(level =>
-      level.ids.find(id => id === levelId)
+    const newLevel = getLevelById(
+      state.lessons,
+      state.currentLessonId,
+      levelId
     );
 
     updateBrowserForLevelNavigation(state, newLevel.url, levelId);
     dispatch(setCurrentLevelId(levelId));
   };
 }
 
+// The user has successfully completed the level and the page
+// will not be reloading.
+export function sendSuccessReport(appType) {
+  return (dispatch, getState) => {
+    const state = getState().progress;
+    const levelId = state.currentLevelId;
+    const currentLevel = getLevelById(
+      state.lessons,
+      state.currentLessonId,
+      levelId
+    );
+    const scriptLevelId = currentLevel.id;
+
+    // The server can determine the user ID.
```

Comment:
nit: can we something like 'a value of 0 indicates to the server to load the user ID from the current session' just to be clear?

Reply:
From perusing the [controller](https://github.com/code-dot-org/code-dot-org/blob/cfa114efdb6cd0d41ebeba9895e7226b93d4752a/dashboard/app/controllers/activities_controller.rb#L22), it appears that the server doesn't even use this parameter anymore...

Reply:
I've added a comment to that effect.